Lambda-list/unparse.lisp

```diff
@@ -0,0 +1,93 @@
+(in-package #:concrete-syntax-tree)
+
+;;;; This file defines an "unparsing" system. Given a parsed lambda list, a
+;;;; lambda list intended for human consumption is returned. The unparse
+;;;; should resemble the originally parsed lambda list, but there is no
+;;;; exactness requirement.
+;;;; This is useful for displaying lambda lists to a user,
+;;;; for example in error reports.
+
+(defgeneric unparse-lambda-list (client lambda-list))
+
+(defgeneric unparse-parameter-group (client parameter-group))
+
+(defgeneric unparse-parameter (client parameter))
+
+;;;
+
+(defmethod unparse-lambda-list (client (lambda-list lambda-list-type))
+  (loop for parameter-group in (children lambda-list)
+        appending (unparse-parameter-group client parameter-group)))
```

Comment:
Maybe omit `&aux` for the same reason specializers and default values are omitted?